crates/syntax/src/syntax_editor/edit_algo.rs

```diff
@@ -73,7 +73,7 @@ pub(super) fn apply_edits(editor: SyntaxEditor) -> SyntaxEdit {
         })
         .all(|(l, r)| {
             get_node_depth(l.target_parent()) != get_node_depth(r.target_parent())
-                || l.target_range().intersect(r.target_range()).is_none()
+                || (l.target_range().end() <= r.target_range().start())
```

Comment:
I considered that but figured I'd rather just be straightforward with what's being checked especially since the ranges are sorted so there's no need to check if `l` starts inside of `r`